shared/services/config/rocket-pool-config.go

```diff
@@ -165,9 +167,13 @@ func LoadFromFile(path string) (*RocketPoolConfig, error) {
 	if err := yaml.Unmarshal(configBytes, &settings); err != nil {
 		return nil, fmt.Errorf("could not parse settings file: %w", err)
 	}
-
+	// now load the networks:
+	networks, err := LoadNetworksFromFile(filepath.Dir(path))
+	if err != nil {
+		return nil, fmt.Errorf("could not load networks file: %w", err)
+	}
```

Comment:
kinda seems like `NewRocketPoolConfig` should handle this. `filepath.Dir(path)` gets stored in `cfg.RocketPoolDirectory` on alloc, and then in `cfg.Deserialize` you can parse the new yml.

Reply:
I've looked into this again and I prefer passing it into New for two reasons... `NewRocketPoolConfig` [calls `NewSmartnodeConfig`](https://github.com/activescott/smartnode/blob/cef7336fc59d22944c1a5de56fbec89b862a18c7/shared/services/config/rocket-pool-config.go#L458) which [loads a list of network options](https://github.com/activescott/smartnode/blob/cef7336fc59d22944c1a5de56fbec89b862a18c7/shared/services/config/smartnode-config.go#L155) so it requires networks to be available already.

There is also a place in [the client creates a fresh config without deserializing](https://github.com/activescott/smartnode/blob/cef7336fc59d22944c1a5de56fbec89b862a18c7/shared/services/rocketpool/client.go#L230) that would require some patching after it is created to even have a valid config. So I think it's more correct to have networks passed in.

Let me know if I'm missing something.

Reply:
Can we meet in the middle and have `NewRocketPoolConfig` parse the files just before calling `NewSmartnodeConfig`?

In both places NewRocketPoolConfig is called it uses the same path as the previous LoadNetworksFromeFile call

Reply:
:) I agree with your edit. 

There is a third place where that `NewRocketPoolConfig` that actually doesn't require reading the networks: https://github.com/activescott/smartnode/blob/cef7336fc59d22944c1a5de56fbec89b862a18c7/shared/services/rocketpool/client.go#L230

After reviewing this, I prefer the way it is now. Let me know what you prefer though. I'm open.